packages/e2e-test-utils/src/click-menu-item.js

```diff
@@ -11,7 +11,7 @@ import { first } from 'lodash';
 export async function clickMenuItem( label ) {
 	const elementToClick = first(
 		await page.$x(
-			`//div[@role="menu"]//span[contains(concat(" ", @class, " "), " components-menu-item__item ")][contains(text(), "${ label }")]`
+			`//div[@role="menu"]//span[contains(concat(" ", @class, " "), " components-menu-item__item ")][text()="${ label }"]`
```

Comment:
I had to update XPath to match the exact label text to fix the e2e testing issue. We always provide full labels in our tests, but not sure how this might affect third-party integrations.

cc/ @ntsekouras, @youknowriad

Reply:
It might affect them yes and I don't think we should make this change as it's a breaking change. 

Are there many tests failing without this change? IMO we should add an extra param in this function about partial/full matching and update the GB tests to use the full match behavior. The default should be the existing partial match.

Reply:
Thanks for the feedback, Nik.

The only Grouping tests were failing since the new menu item uses a block name, and it will match "Lock Group" before matching the "Group" item.

I will create a separate PR for the "partial/full" match argument.

Reply:
I think this change is a dev package, so I wouldn't mind a breaking change personally if we just document it in CHANGELOG.md of the package.

Reply:
Created separate PR for this change - #39274.